ntbs-service/Pages/Alerts/TransferRequest.cshtml.cs

```diff
@@ -0,0 +1,181 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using ntbs_service.DataAccess;
+using ntbs_service.Models;
+using ntbs_service.Models.Entities;
+using ntbs_service.Models.Enums;
+using ntbs_service.Models.FilteredSelectLists;
+using ntbs_service.Models.ReferenceEntities;
+using ntbs_service.Pages.Notifications;
+using ntbs_service.Services;
+
+namespace ntbs_service.Pages.Alerts
+{
+    public class TransferRequestModel : NotificationModelBase
+    {
+        private readonly IAlertRepository alertRepository;
+        private readonly IAlertService alertService;
+        private readonly IReferenceDataRepository _referenceDataRepository;
+        public ValidationService ValidationService { get; set; }
+
+        [BindProperty]
+        public TransferAlert TransferAlert { get; set; }
+
+        [BindProperty]
+        public int AlertId { get; set; }
+        public SelectList TbServices { get; set; }
+        public SelectList CaseManagers { get; set; }
+        public SelectList Phecs { get; set; }
+
+        public TransferRequestModel(
+            INotificationService notificationService,
+            IAlertService alertService, 
+            IAlertRepository alertRepository,
+            IAuthorizationService authorizationService,
+            INotificationRepository notificationRepository,
+            IReferenceDataRepository referenceDataRepository) : base(notificationService, authorizationService, notificationRepository)
+        {
+            this.alertService = alertService;
+            this.alertRepository = alertRepository;
+            this._referenceDataRepository = referenceDataRepository;
+            ValidationService = new ValidationService(this);
+        }
+
+        public async Task<IActionResult> OnGetAsync()
+        {
+            Notification = await NotificationRepository.GetNotificationAsync(NotificationId);
+            await AuthorizeAndSetBannerAsync();
+            var alert = await alertRepository.GetOpenAlertByNotificationIdAndTypeAsync(NotificationId, AlertType.TransferRequest);
+            if (alert != null)
+            {
+                TransferAlert = (TransferAlert)alert;
+                return Partial("_TransferPendingPartial", this);
+            }
+            else
+            {
+                TransferAlert = new TransferAlert();
+            }
+
+            // Check edit permission and redirect if not allowed
+            if (!HasEditPermission)
+            {
+                return RedirectToPage("/Notifications/Overview", new { NotificationId });
+            }
+
+            await SetDropdownsAsync();
+
+            return Page();
+        }
+
+        public async Task<IActionResult> OnPostConfirmAsync()
+        {
+            Notification = await NotificationRepository.GetNotificationAsync(NotificationId);
+            await GetRelatedEntities();
+            ModelState.Clear();

```

Comment:
Need to clear the model (as it will be invalid from a validation attempt prior to getting the related entities) so that we can revalidate it on the next line